src/test/groovy/graphql/execution/reactive/CompletionStageMappingOrderedPublisherTest.groovy

```diff
@@ -0,0 +1,280 @@
+package graphql.execution.reactive
+
+import graphql.TestUtil
+import graphql.execution.pubsub.CapturingSubscriber
+import io.reactivex.Flowable
+import org.awaitility.Awaitility
+import org.reactivestreams.Publisher
+import spock.lang.Specification
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionException
+import java.util.concurrent.CompletionStage
+import java.util.function.Function
+
+class CompletionStageMappingOrderedPublisherTest extends Specification {
```

Comment:
The tests for the two classes kinda repeat themselves - but I figuered that even though they inherent from each other - the tests should stress them the same and make no assumptions on how they are implemented

With even more work maybe the tests become common ones use `where:` but I didnt go that far - copy pasta rather

Reply:
And - no I ended up refactoring so that the same tests are run for both implementations - one in order and one not in order